analysis/analysis.py

```diff
@@ -446,6 +450,203 @@ def test_word_time_relations(self):
+                                                                        'a short story.'])
+        exclusive_responses = total_reading_time_exclusive(wanted_dict_ad_avg, wanted_dict_ss_avg)
+        self.assertEqual(exclusive_responses, {'This is an ad.': {},
+                                               'This is actually a short story.': {}})
+
+        view_times_per_response = compute_view_time_per_response(
+            self.default_student_data)
+        wanted_dict_ad_avg = build_average_dict(view_times_per_response['This is an ad.'])
+        wanted_dict_ss_avg = build_average_dict(view_times_per_response['This is actually '
+                                                                        'a short story.'])
+        exclusive_responses = total_reading_time_exclusive(wanted_dict_ad_avg, wanted_dict_ss_avg)
+        self.assertEqual(exclusive_responses, {'This is an ad.': {},
+                                               'This is actually a short story.': {}})
+
+
+def compute_view_time_per_response(student_data):
+    """
+    Compute the total reread times for each response
+    :param student_data: list, student response dicts
+    :return:
+    wanted_dict_ad - Dictionary ('ad context response in lowercase': [total reading times for
+        each user])
+    wanted_dict_ss - Dictionary ('short story context response in lowercase': [total reading
+        times for each user])
+    """
+
+    wanted_dict_ad = defaultdict(list)
+    wanted_dict_ss = defaultdict(list)
+
+    for entry in student_data:
+        if entry['question'] == 'In one word, how does this text make you feel?':
+            response = entry['response'].lower()
+            views = sum(entry['views'])
+
+            if entry['context'] == 'This is an ad.':
+                wanted_dict_ad[response].append(views)
+
+            elif entry['context'] == 'This is actually a short story.':
+                wanted_dict_ss[response].append(views)
+
+    return {'This is an ad.': wanted_dict_ad, 'This is actually a short story.': wanted_dict_ss}
+
+
+def get_common_elements(ad_response_times, ss_response_times):
+    """
+    make a dictionary of the common elements between two dictionaries and averages them
+    :param ad_response_times: the first dictionary to compare
+    :param ss_response_times: the second dictionary to compare
+    :return: one dictionary
+    """
+    # Start making a combined dictionary that does not separate based on context
+    wanted_dict_combined = {}
+    for word in ad_response_times.keys():
+        if word not in wanted_dict_combined.keys():
+            wanted_dict_combined[word] = ad_response_times[word]
+
+    # Continue building the combined dictionary with short story dictionary
+    for word in ss_response_times.keys():
+        # Average current response with new data
+        if word in wanted_dict_combined.keys():
+            wanted_dict_combined[word] = (wanted_dict_combined[word] + ss_response_times[word]) / 2
```

Comment:
can you name the result of the calculation and then store that? would be clearer to read.